From 97552d275594169c1287237aa4b139d41f60dc7f Mon Sep 17 00:00:00 2001
From: Vendula Poncova <vponcova@redhat.com>
Date: Sun, 1 Mar 2020 10:54:37 +0100
Subject: [PATCH] Use the latest kernel version list (#1807252)

The kernel version list changes during the installation, so create the DBus task
for the bootloader configuration with the latest list right before we run it.

Resolves: rhbz#1807252
---
 pyanaconda/installation.py | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/pyanaconda/installation.py b/pyanaconda/installation.py
index 7509f79bf..e8b147bb6 100644
--- a/pyanaconda/installation.py
+++ b/pyanaconda/installation.py
@@ -23,6 +23,7 @@ from pyanaconda.modules.common.constants.objects import BOOTLOADER, SNAPSHOT, FI
 from pyanaconda.modules.common.constants.services import STORAGE, USERS, SERVICES, NETWORK, SECURITY, \
     LOCALIZATION, TIMEZONE, BOSS
 from pyanaconda.modules.common.structures.requirement import Requirement
+from pyanaconda.modules.common.task import sync_run_task
 from pyanaconda.payload.livepayload import LiveImagePayload
 from pyanaconda.progress import progress_message, progress_step, progress_complete, progress_init
 from pyanaconda import flags
@@ -322,9 +323,13 @@ def _prepare_installation(payload, ksdata):
     bootloader_proxy = STORAGE.get_proxy(BOOTLOADER)
     bootloader_install = TaskQueue("Bootloader installation", N_("Installing boot loader"))
 
-    if not payload.handles_bootloader_configuration:
+    def configure_bootloader():
         boot_task = bootloader_proxy.ConfigureWithTask(payload.kernel_version_list)
-        bootloader_install.append_dbus_tasks(STORAGE, [boot_task])
+        sync_run_task(STORAGE.get_proxy(boot_task))
+
+    if not payload.handles_bootloader_configuration:
+        # FIXME: This is a temporary workaround, run the DBus task directly.
+        bootloader_install.append(Task("Configure the bootloader", configure_bootloader))
 
     bootloader_install.append_dbus_tasks(STORAGE, [bootloader_proxy.InstallWithTask()])
     installation_queue.append(bootloader_install)
-- 
2.24.1

